smart-contarct.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.7;
+
+contract Casino {
+    uint private balance;
+    uint public minVal;
+    uint public maxVal;
+    uint public Cap;
+    address public owner;
+    uint public payments;
+    uint public profit;
+
+    constructor() {
+        owner = msg.sender;
+    }  
+
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Access restricted, not an owner");
+        _;
+    }
+
+    function setMinTopUp(uint  val) public onlyOwner() {
+        minVal = val;
+    }
+
+    function setMaxTopUp(uint  val) public onlyOwner() {
+        maxVal = val;
+    }
+    
+    function setCap(uint val) public onlyOwner() {
+        Cap = val;
+    }
+
+    function topUp() public payable {
+        payments = msg.value;
+        if( payments >= Cap) {

```

Comment:
You need to require value to be more than min value
Less than max value
Also you need to keep all round values in one var, you are overriding them